lib/stm32/l1/flash.c

```diff
@@ -230,5 +230,50 @@ void eeprom_program_words(uint32_t address, uint32_t *data, int length_in_words)
 	flash_lock_pecr();
 }
 
+/** @brief Erase a page (256 Byte) of program memory (flash)
+ *
+ * @param[in] page_address must point to flash space, no checking!
+ */
+void flash_erase_page(uint32_t address)
+{
+	while (FLASH_SR & FLASH_SR_BSY);
+	FLASH_PECR |= (FLASH_PECR_ERASE | FLASH_PECR_PROG);
+	*(uint32_t*)address = 0;
+	while (FLASH_SR & FLASH_SR_BSY);
+	FLASH_PECR &= ~(FLASH_PECR_ERASE | FLASH_PECR_PROG);
+}
+
+/** @brief Program memory: write half page to flash
+ *
+ * A half page is 32 words (128 byte).
+ *
+ * Half Page Write is possible only from SRAM, from external memory or from
+ * bank 1 to bank 2 and vice versa. Force into SRAM with the section attribute.
+ *
+ * @param[in] address must point to flash space, no checking!
+ * @param[in] data pointer to 128 bytes of data to write
+ */
+__attribute__((section(".data#")))
```

Comment:
Right. Still it would be a good reference to have this function in the library - what about just adding to the documentation that this needs to be placed into SRAM  or according to the rules (externam memory, other bank...) in project specific linker script?